### PR TITLE
Fix fallback to local PoW, add tls as default feature

### LINKED ISSUE #147: Fix fallbackToLocalPow
## Description

Client with `"localPow":false,"fallbackToLocalPow":true` failed to reattach a transaction
```
2023-03-29 11:30:14 (UTC) iota_wallet::account::operations::transaction::submit_transaction [94mDEBUG[0m [TRANSACTION] send_payload
2023-03-29 11:30:14 (UTC) iota_client::node_manager::http_client     [94mDEBUG[0m GET: 56 ms for 200 OK https://api.testnet.shimmer.network/api/core/v2/tips
2023-03-29 11:30:14 (UTC) iota_wallet::message_interface::message_handler [94mDEBUG[0m Response: Error(Client(HealthyNodePoolEmpty))
```
because of fallbackToLocalPow it should still work and not return an error

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Fix fallback to local PoW, add tls as default feature

## Links to any relevant issues

Fixes https://github.com/iotaledger/iota-sdk/issues/147

## Type of change

- Bug fix (a non-breaking change which fixes an issue)

## How the change has been tested

Running the post block example with `.with_local_pow(false)`

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: sdk/src/client/node_api/core/routes.rs [hunk 1]
```diff
@@ -240,7 +240,7 @@ impl Client {
                 if let Error::Node(e) = e {
                     let fallback_to_local_pow = self.get_fallback_to_local_pow();
                     // hornet and bee return different error blocks
-                    if (e == *"No available nodes with remote Pow"
+                    if (e == *"no available nodes with remote Pow"
```
Comment (thibault-martinez) [inline sdk/src/client/node_api/core/routes.rs:243]: scam

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Conflict, sorry

File: sdk/src/client/node_manager/mod.rs [hunk 2]
```diff
@@ -139,6 +139,10 @@ impl NodeManager {
         // remove disabled nodes
         nodes_with_modified_url.retain(|n| !n.disabled);
 
+        if nodes_with_modified_url.is_empty() {
```
Comment (thibault-martinez) [inline sdk/src/client/node_manager/mod.rs:142]: Isn't that the same check as after?

Comment (Thoralf-M) [reply sdk/src/client/node_manager/mod.rs:142]: Good catch

**Review by thibault-martinez — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

**Pull request merged**